### PR TITLE
FEAT: Added dark theme

### PR DESCRIPTION
**Pull request by zoyatanweer**
Added theme option
Files to check : 
- Context/theme/Theme-Context.jsx

Made minor fixes:
- Made wishlist icon bigger and made it color pink when clicked on card.
- made active link color as primary color.

### CODE REVIEW

**Review by humzahasan — APPROVED** [review]
- Code changes are good with descriptive variables and functions
- UI looks very clean and minimalistic

Good to merge!

**Review by srishtigautam21 — COMMENTED** [review]
in your UI after i change to dark theme and click on login icon the site reloads due to which the dark theme changes to light theme

**Comment by zoyatanweer** [comment]
> in your UI after i change to dark theme and click on login icon the site reloads due to which the dark theme changes to light theme

Thanks for pointing it out Srishti! Will look into it.

File: src/index.css [hunk 1]
```diff
@@ -33,7 +33,8 @@ a:not([class]) {
 
 a {
   text-decoration: none;
-  color: inherit;
+  /* color: inherit; */
```
Comment (zoef-yt) [inline src/index.css]: Remove commented code.

File: src/components/wishlist/Wishlist.jsx [hunk 2]
```diff
@@ -1,81 +1,106 @@
...
-                    {<AddToWishlist className="far fa-heart" />}
-                  </button>
-                  <button className="share-btn">{<ShareIcon />}</button>
+                  <span className="hide dismiss-btn">X</span>
+                  <span className="card-rating rated-star">
+                    {<RatingIcon className="rated rated-star rating-size" />}
+                    {ratings}
+                  </span>
+                  <div className="hide overlay">
+                    {inStock ? "" : "out of stock"}
```
Comment (zoef-yt) [inline src/components/wishlist/Wishlist.jsx:61]: here you can just do `{!inStock  && "out of stock"}`

File: src/components/products/Products.jsx [hunk 3]
```diff
@@ -192,7 +198,8 @@ const Products = () => {
         </div>
         {/* <!-- flex-filter --> */}
       </div>
-    </>
+    </div>
+    // </>
```
Comment (zoef-yt) [inline src/components/products/Products.jsx]: remove commented code.

File: src/components/products/Products.jsx [hunk 4]
```diff
@@ -101,7 +100,8 @@ const Products = () => {
   };
 
   return (
-    <>
+    // <>
```
Comment (zoef-yt) [inline src/components/products/Products.jsx]: remove commented code.

File: src/components/products/Products.css [hunk 5]
```diff
@@ -181,13 +189,10 @@
 .slider::-webkit-slider-thumb {
   -webkit-appearance: none;
   appearance: none;
-  /* width: 25px; */
   width: 20px;
-  /* height: 25px; */
   height: 20px;
```
Comment (zoef-yt) [inline src/components/products/Products.css]: use rem.

File: src/components/products/Products.css [hunk 6]
```diff
@@ -60,7 +61,8 @@
   cursor: pointer;
 }
 .filter-item:hover {
-  color: var(--navigation-color);
+  /* color: var(--navigation-color); */
```
Comment (zoef-yt) [inline src/components/products/Products.css]: remove commented code.

File: src/components/products/Filter.jsx [hunk 7]
```diff
@@ -282,7 +285,9 @@ const Filter = () => {
           </>
         </ul>
       </div>
-    </>
+    </div>
+
+    // </>
```
Comment (zoef-yt) [inline src/components/products/Filter.jsx]: remove commented code.

File: src/components/products/Filter.jsx [hunk 8]
```diff
@@ -1,11 +1,14 @@
...
 import { useFilter } from "../../Context/Filter-Context";
+import { useTheme } from "../../Context/Theme/Theme-Context";
 
 const Filter = () => {
+  const { theme } = useTheme();
   const { FilterState, FilterDispatch } = useFilter();
 
   return (
-    <>
+    // <>
```
Comment (zoef-yt) [inline src/components/products/Filter.jsx]: remove commented code.

File: src/components/header/Header.jsx [hunk 9]
```diff
@@ -56,7 +73,7 @@ const Header = () => {
           <div className="single-action-btn">
             <Link to="/cart">
               <NavLinkItems
-                className="single-action-btn"
+                // className="single-action-btn"
```
Comment (zoef-yt) [inline src/components/header/Header.jsx]: remove commented code.

File: src/components/header/Header.jsx [hunk 10]
```diff
@@ -2,34 +2,45 @@ import React from "react";
...
 import { useCart } from "../../Context/Cart/Cart-Context";
 import { CartProvider } from "../../Context/Cart/Cart-Context";
+import { ThemeProvider, useTheme } from "../../Context/Theme/Theme-Context";
 
 const Header = () => {
   const { wishlist } = useWishlist();
   const { cart, initialValue } = useCart();
+  const { theme, toggleTheme } = useTheme();
+  const navLinkClass = ({ isActive }) =>
+    `link ${isActive ? "link-active" : ""}`;
```
Comment (zoef-yt) [inline src/components/header/Header.jsx:24]: Here you can try `${isActive && "link-active"}`

File: src/components/header/Header.css [hunk 11]
```diff
@@ -60,12 +75,3 @@
   right: 0;
   left: 18px;
```
Comment (zoef-yt) [inline src/components/header/Header.css]: Use rem

File: src/components/header/Header.css [hunk 12]
```diff
@@ -26,7 +25,8 @@
   flex-direction: column;
   padding: 0;
   margin: 0;
-  background-color: var(--background-color);
+  /* background-color: var(--background-color); */
```
Comment (zoef-yt) [inline src/components/header/Header.css]: remove commented code.

File: src/components/cart/CartTotal.jsx [hunk 13]
```diff
@@ -1,9 +1,10 @@
-// import { CartData } from "./CartData";
 import cartSection from "../../Assets/Images/cartSection.png";
 import { useEffect, useState } from "react";
 import { useCart } from "../../Context/Cart/Cart-Context";
+import { useTheme } from "../../Context/Theme/Theme-Context";
```
Comment (zoef-yt) [inline src/components/cart/CartTotal.jsx:4]: Suggestion you can make index.js file for context too.

File: src/components/cart/Cart.jsx [hunk 14]
```diff
@@ -3,108 +3,124 @@ import "./Cart.css";
...
+                    className="card-img img-horizontal"
+                    src={imageSrc}
+                    alt={name}
+                  />
+                  {hasOffer && (
+                    <span className="card-badge">{badgeMessage}</span>
+                  )}
+                  <span className="hide dismiss-btn">X</span>
+                  <div className="hide overlay">
+                    {inStock ? "" : "out of stock"}
```
Comment (zoef-yt) [inline src/components/cart/Cart.jsx:51]: You can try `{!inStock &&"out of stock"}`

**Review by zoef-yt — CHANGES_REQUESTED** [review]

**Review by zoef-yt — APPROVED** [review]
Do the suggested changes and you are good to merge. Nice work Zoya!.

**Comment by zoyatanweer** [comment]
Thankyou all for your valuable feedbacks.

**Pull request merged**